cpgserver/src/main/scala/io/shiftleft/cpgserver/route/CpgRoute.scala

```diff
@@ -103,29 +173,36 @@ final class CpgRoute(cpgProvider: CpgProvider, cpgQueryExecutor: ServerAmmoniteE
 
 object CpgRoute {
 
-  def apply(cpgProvider: CpgProvider, ammoniteExecutor: ServerAmmoniteExecutor)(
-      implicit httpErrorHandler: HttpErrorHandler): CpgRoute = {
-    new CpgRoute(cpgProvider, ammoniteExecutor)
-  }
-
+  /* Response types */
   final case class ApiError(error: String)
   final case class CreateCpgRequest(files: List[String])
   final case class CreateCpgResponse(uuid: UUID)
   final case class CreateCpgQueryRequest(query: String)
   final case class CreateCpgQueryResponse(uuid: UUID)
   final case class CpgOperationResponse(ready: Boolean, result: Option[String] = None, error: Option[String] = None)
 
+  /* Error handling types */
+  final case class FileNameMissingException(message: String) extends RuntimeException(message)
+  final case class FileSizeException(message: String) extends RuntimeException(message)
+
+  /* Entity decoders. */
   private[route] implicit val decodeCpgRequest: EntityDecoder[IO, CreateCpgRequest] = jsonOf
   private[route] implicit val decodeCpgQueryRequest: EntityDecoder[IO, CreateCpgQueryRequest] = jsonOf
 
+  def apply(cpgProvider: CpgProvider, ammoniteExecutor: ServerAmmoniteExecutor, config: ServerFilesConfiguration)(
+      implicit httpErrorHandler: HttpErrorHandler,
+      cs: ContextShift[IO]): CpgRoute = {
+    new CpgRoute(cpgProvider, ammoniteExecutor, config)
+  }
+
   object CpgHttpErrorHandler extends HttpErrorHandler {
     private val logger = LoggerFactory.getLogger(this.getClass)
 
     override def handle(routes: HttpRoutes[IO]): HttpRoutes[IO] =
       HttpErrorHandler(routes) {
         case _: MessageBodyFailure =>
           BadRequest(ApiError("Invalid payload. Please check that the payload is formatted correctly.").asJson)
-        case ex =>
+        case NonFatal(ex) =>
```

Comment:
non-exhaustive check doesn't complain?

Reply:
This takes a `PartialFunction` :)

Reply:
Aha, so I'm wrong! Nice spot, will fix it.